pkg/cmd/roachtest/decommission.go

```diff
@@ -516,40 +534,51 @@ func runDecommissionRandomized(ctx context.Context, t *test, c *cluster) {
 			t.Fatal(err)
 		}
 
-		// Check that even though two nodes are decommissioned, we still see
-		// them (since they remain live) in `node ls`.
+		// Check that the two decommissioned nodes vanish from `node ls` (since they
+		// will not remain live as the cluster refuses connections from them).
 		{
-			runNode = h.getRandNode()
-			t.l.Printf("checking that `node ls` (from n%d) shows all nodes\n", runNode)
-			o, err := execCLI(ctx, t, c, runNode, "node", "ls", "--format=csv")
-			if err != nil {
-				t.Fatalf("node-ls failed: %v", err)
-			}
-			exp := [][]string{{"id"}}
-			for i := 1; i <= c.spec.NodeCount; i++ {
-				exp = append(exp, []string{strconv.Itoa(i)})
-			}
-			if err := h.matchCSV(o, exp); err != nil {
+			if err := retry.WithMaxAttempts(ctx, retry.Options{}, 50, func() error {
```

Comment:
I assume this retry loop here, above and below were all needed for the same reason? Propagation delays to the invoking node?

Reply:
The decommissioned nodes have to lose their liveness first.